### ISSUE TITLE
Roadmap for tools in 0.5.0

### ISSUE THREAD

**Issue by liquidev** [liquidev/netcanv#47]
- [x] Tools
  - [x] Selection
    - Keyboard shortcuts
      - [x] Ctrl+C and Ctrl+V
      - [x] Delete
    - ~~Cursors~~
      Blocked by winit 0.26.0; resize cursors are swapped on 0.25.0 and I don't want to release a buggy product.
    - ~~Proportional selection (Shift)~~
      - Maybe also another key for integer scaling. This isn't something most drawing apps have, but I think it would be a nice addition.
      - After reevaluating the amount of code changes this would need, I have decided to postpone this.
  - [x] Brush
    - ~~Custom cursor~~
      Doesn't seem like winit exposes an API for custom cursors, so that's out of the question.
- [x] Protocol support for tools
  - Synchronizing tool state across clients
    - [x] Tool selection packets
    - [x] Brush packets
    - [x] Selection packets
- [x] Config setting for moving the toolbar to the right
  - How do we expose this to the user? A nice way would be simply having a handle for dragging the toolbar over, but that's out of scope for this update as it would require a major rewrite of how NetCanv does panels. Thus, I think maybe we should keep this hidden for now.

Known bugs:

- [x] The brush guide does not scale with the zoom level.
- [x] Selection is not draggable if its size is negative.
- [x] Pasted selection size is not limited to 1024x1024 like all other selections are.

**Comment by liquidev** [comment]
All seems to be done! We'll still need to conduct some bug testing but all the basics have been completed already.
